tests/unit/amo/components/TestCollection.js

```diff
@@ -0,0 +1,445 @@
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({
+      location: newLocation,
+      params: { slug: newSlug, user },
+    });
+
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(fakeDispatch, fetchCollection({
+      errorHandlerId: errorHandler.id,
+      page,
+      slug: newSlug,
+      user,
+    }));
+  });
+
+  it('dispatches fetchCollectionPage when page has changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const page = 123;
+    const location = { query: {} };
+    const newLocation = { query: { page } };
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    const wrapper = renderComponent({
+      errorHandler,
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({ location: newLocation });
+
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(fakeDispatch, fetchCollectionPage({
+      errorHandlerId: errorHandler.id,
+      page,
+      slug,
+      user,
+    }));
+  });
+
+  it('renders a collection', () => {
+    const store = dispatchClientMetadata().store;
+
+    const collectionAddons = createFakeCollectionAddons();
+    const collectionDetail = createFakeCollectionDetail();
+
+    store.dispatch(loadCollection({
+      addons: collectionAddons,
+      detail: collectionDetail,
+    }));
+
+    const wrapper = renderComponent({ store });
+    expect(wrapper.find('.Collection-wrapper')).toHaveLength(1);
+    expect(wrapper.find(AddonsCard)).toHaveLength(1);
+    expect(wrapper.find(Paginate)).toHaveLength(1);
+  });
+
+  it('renders loading indicator on add-ons when fetching next page', () => {
+    const store = dispatchClientMetadata().store;
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'user-id-or-name';
+
+    // User loads the collection page.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const wrapper = renderComponent({
+      errorHandler,
+      params: { slug, user },
+      store,
+    });
+    expect(wrapper.find(AddonsCard)).toHaveProp('loading', false);
+
+    // User clicks on 'next' pagination link.
+    store.dispatch(fetchCollectionPage({
+      errorHandlerId: errorHandler.id,
+      page: 2,
+      slug,
+      user,
+    }));
+
+    wrapper.setProps(mapStateToProps(store.getState()));
```

Comment:
Why do we have to `setProps` in so many places? Is `shallow` missing certain lifecycle stuff? I thought there was a special option to enable that? Or we could use `mount`? I'd like to approximate dispatches as closely as possible if we can and it's not a huge pain.

Reply:
I think it's `lifecycleExperimental: true` we want to use, like here: https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestLandingPage.js#L37

Reply:
`shallow` does not trigger lifecycle methods on purpose. We use `setProps()` quite often actually, and I thought that was the way to go since it does exactly what we want here, that is triggering `componentWillReceiveProps()`.

Reply:
Note that here I use a `store` and dispatch the proper actions so isn't it "as closely as possible" already? I am open to suggestions, please let me know.

Reply:
Yeah I think you're right, never mind me 😄